### ISSUE TITLE
Add a «Swap Layers» patterned after «Copy Layer to Layer»

### ISSUE THREAD

**Issue by aminabedi68** [fontforge/fontforge#4361]
Hi
I'm not sure it's totally impossible or just not working in my version(20200314 on windows10).
I'm working on a color font with two layers in fore and back, and I have to switch them by editing the source with a text editor, because I can't do that inside fontforge (layer tab in font information has some options, but it's not working).

**Comment by ctrlcctrlv** [comment]
Hello!

We don't yet support color fonts, but I plan to implement it. So how are you working on one? Please detail your workflow and the problem in greater detail

**Comment by aminabedi68** [comment]
yes, I know, I've been waiting for it [since ](https://github.com/fontforge/fontforge/issues/677)2013 :)
I preparing the layers(back & fore) in fontforge, generate the font , switch layers with notpad++, generate another layer, and give them to [opentype-svg](https://github.com/adobe-type-tools/opentype-svg). although COLR/CPAL has a greater advantage to create color-**variable** fonts, but I can't find a simple solution or opensource tool for that.
I realized that I can use notpad++ to switch layers, it's so easy but before that, I was switching all glyphs manually!
and you should consider that we can't use a built script to do that(or we can? I'm not sure)

**Comment by ctrlcctrlv** [comment]
Sorry I'm still confused. What tool converts FontForge's layers into color information? And as to 2013, I've only been planning our COLR/CPAL support for six months or so. I haven't gotten to it as quickly as I'd have liked to. Thanks for your patience. :-)

**Comment by skef** [comment]
If you're not using references then it would be very easy to do this sort of switch with a python script. If you are then it's not quite as easy but still doable.

**Comment by skef** [comment]
@ctrlcctrlv I think he's doing development in FontForge with various layers and then exporting the layers one at a time into separate fonts. The tool he links to appears to take one-font-per-layer as input and then synthesizes a color font. See https://github.com/adobe-type-tools/opentype-svg#step-1 for example.

**Comment by ctrlcctrlv** [comment]
Thanks @skef, I think I get it now. Yes, a script is best bet for this as a quick work around. Although, we ought to consider a swap layers function in the UI. Like the existing «Copy Layer to Layer».

**Comment by aminabedi68** [comment]
thank you all, @skef is right

**Comment by skef** [comment]
@aminabedi68 Here's a script that doesn't handle references:

```
#!/usr/bin/env python

import sys
import fontforge

fn = sys.argv[1]

f = fontforge.open(fn)
layers = []
for l in f.layers:
    layers.append(l)
f.close()

for ln in layers:
    nfn = fn.replace('.sfd', '_' + ln + '.otf')
    f = fontforge.open(fn)
    f.selection.all()
    for g in f.selection.byGlyphs:
        g.foreground = g.layers[ln]
    f.generate(nfn)
    f.close()
```

When passed the name of an sfd file it should generate an OTF for each layer, with the layer in the name. 

Is this roughly what you need? Does your font use references?

**Comment by aminabedi68** [comment]
thank you @skef ! it's exactly what I want! :)
no, it's a font icon, there is no referenced glyph in the font.

**Comment by skef** [comment]
FontForge: "Where even when we give the user exactly what they need, it seems like the issue should stay open!"

**Comment by skef** [comment]
@aminabedi68 That wasn't a passive-aggressive dig on you -- I believe @ctrlcctrlv wants to keep this open to track a potential feature to add to the main program.

**Comment by ctrlcctrlv** [comment]
Yes, I do.
